libraries/Microsoft.Bot.Builder/PrivateConversationState.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Handles persistence of a conversation state object using the conversation ID and From ID as part of the key.
+    /// </summary>
+    public class PrivateConversationState : BotState
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrivateConversationState"/> class.
+        /// </summary>
+        /// <param name="storage">The storage provider to use.</param>
+        public PrivateConversationState(IStorage storage)
+            : base(storage, nameof(PrivateConversationState))
+        {
+        }
+
+        /// <summary>
+        /// Gets the key to use when reading and writing state to and from storage.
+        /// </summary>
+        /// <param name="turnContext">The context object for this turn.</param>
+        /// <returns>The storage key.</returns>
+        protected override string GetStorageKey(ITurnContext turnContext) => $"{turnContext.Activity.ChannelId}/conversations/{turnContext.Activity.Conversation.Id}/users/{turnContext.Activity.From.Id}";
```

Comment:
Are we worried about null checking here? If this were (For example) called from the proactive case, this would probably fail with Null Reference Exceptions. I'm on the fence...